setup.py

```diff
@@ -410,9 +421,11 @@ def _compile_webhost(webhost_dir: pathlib.Path):
 
     def run(self):
         # Prepare webhost
-        zip_path = self._download_webhost_zip(self.webhost_version)
+        zip_path = self._download_webhost_zip(self.webhost_version,
+                                              self.use_branch)
         self._create_webhost_folder(self.webhost_dir)
-        self._extract_webhost_zip(version=self.webhost_version,
+        version = self.use_branch or self.webhost_version
+        self._extract_webhost_zip(version=version.replace('/', '-'),
```

Comment:
add null check in case version is None?

Reply:
We dont need a null check here.. If branch_name is none then it should use webhost_version which can never be none since we default it to some value